### ISSUE TITLE
[REQ][Python] UUID support

### ISSUE THREAD

**Issue by igorbanfi** [OpenAPITools/openapi-generator#19485]
Currently in python generator UUID are mapped to str. Are there any plans to enable generation of type UUID?

https://github.com/OpenAPITools/openapi-generator/blob/dfc381c22fde85be83681199d39f754efe04061e/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java#L160C9-L160C10

**Comment by c99SRS** [comment]
Hi @igorbanfi you can assign this to me under hacktobestfest tag. Will research on this

**Comment by igorbanfi** [comment]
Hello, @c99SRS I am not a maintainer of openapi-generator and I am unable to assign issues. You will have to contact the maintainers of this repo.

**Comment by to-bee** [comment]
To make this work, the most critical change is reordering the type check conditions. The UUID type must be checked before the string type, as UUID is a subtype of string (with the format: UUID). This ensures the UUID condition is correctly applied.

`AbstractPythonCodegen.java`:
<img width="436" alt="image" src="https://github.com/user-attachments/assets/68d01587-9ce0-428f-855e-0bf65b0372b6" />

`DefaultCodegen.java`:
<img width="1305" alt="image" src="https://github.com/user-attachments/assets/27e24bc7-713a-4440-bfb1-d50235f70a9a" />

`api_client.mustache`:
<img width="489" alt="image" src="https://github.com/user-attachments/assets/2b34d9b3-6633-4de8-8682-fefe2129a1f1" />

**Comment by wing328** [comment]
@to-bee I wonder if you can file a PR with the suggested change 🙏

**Comment by OptimeeringBigya** [comment]
I wish some one would do the same for mapping  `date-time` Openapi spec to `timedelta` python.

**Comment by EspenHa** [comment]
This can be closed now, it is fixed in 7.15

**Comment by ArtemIlinArammeem** [comment]
@wing328 this issue is not mentioned in [release notes](https://github.com/OpenAPITools/openapi-generator/releases/tag/v7.15.0). Is it released?

**Comment by wing328** [comment]
@ArtemIlinArammeem the release note only shows highlights (some of the PRs).

There's a link to all merged PRs included in the release.

**Comment by alexiswl** [comment]
This is causing some problems in my IDE:

Where the `job_id`  parameter in the `api_instance.get_job` method is of type `Annotated[UUID, Field(description="The ID of the job.")],`


<img width="1008" height="625" alt="Image" src="https://github.com/user-attachments/assets/0f8d12fb-4097-4e6c-bd7a-b80ccbaecf96" />

Solutions on user end is to set top-level job definition to be a union type, compared to the screenshot above, we no longer get the error. 

<img width="685" height="623" alt="Image" src="https://github.com/user-attachments/assets/c2ebee48-1780-40de-bef6-d124d102b217" />

However, I wonder if there is a solution on the openapi-generator end (that would save me having to update all my method types).
From [this discussion](https://github.com/pydantic/pydantic/discussions/7023), could UUIDs instead be typed with `UUID4 | Annotated[str, AfterValidator(lambda x: uuid.UUID(x, version=4))]`, I would be keen to know if the AfterValidator from pydantic v2 works and actually validates the parameter or this is for models only. 

Alternatively, you can 'cast' your string variable as a UUID4 object, with the following

```
api_instance.get_job(cast(UUID4, cast(object, job_id)))
```

**Comment by alexiswl** [comment]
@to-bee  in [your statement](https://github.com/OpenAPITools/openapi-generator/issues/19485#issuecomment-2569940124) _as UUID is a subtype of string_ I'm not sure this is correct.

```python3
from uuid import UUID, uuid4

x = uuid4()  # x = UUID('1d111d69-48b1-47f0-8501-47102e37c771')


isinstance(x, UUID)
# True (x is of type UUID)

isinstance(x, str)
# False (x is not an instance of type str)

issubclass(UUID, str)
# False (UUID is not a subclass of str)

issubclass(str, UUID)
# False (str is not subclass of UUID)
```

Maybe I'm missing something here?

**Comment by EspenHa** [comment]
@alexiswl I think what was meant by "UUID is a subtype of string" is that this is how it is encoded in openapi specification.
I believe this is because there is not native UUID type in json.

Regarding your code, you should be able to only modify the function as follows, without changing the function definition:

```python
import uuid

api_instance.get_job(uuid.UUID(job_id)) # convert str to UUID
```

**Comment by alexiswl** [comment]
> Regarding your code, you should be able to only modify the function as follows, without changing the function definition:

Thanks @EspenHa, that does at least provide the assertion that the job_id variable is in-fact a UUID. <br><br>

Another change I've noticed is that the response body attributes from the sdk are also now UUID objects (rather than strings) for our project. This means that they have different methods and need to be handled differently too. 

**project_id attribute of the object is now a UUID object, was previously a string**
<img width="911" height="228" alt="Image" src="https://github.com/user-attachments/assets/0bc095ce-a16d-44d5-9db8-52bebae4e7b0" />

**project_id attribute has UUID methods, rather than str methods**

<img width="770" height="369" alt="Image" src="https://github.com/user-attachments/assets/66856812-e58b-4e17-bad8-41d2c2a10bc7" />

**Comment by EspenHa** [comment]
Yes, this is to be expected. You can convert back to string by using `str(.project_id)`

**Comment by rain87** [comment]
this is quite a breaking change; any idea how do I preserve the old behavior for the backward compatibility? I tried
```
  --type-mappings uuid=str \
  --import-mappings uuid= \
```
but that didn't work

for now, I ended up just stripping `uuid` format specification from the opanapi doc
```
jq 'walk(
    if type == "object" and .format? == "uuid"
    then del(.format)
    else .
    end
)'
```
but this isn't very nice solution
